drivers/gpio/gpio_gd32.c

```diff
@@ -55,6 +55,7 @@ struct gpio_gd32_data {
 	sys_slist_t callbacks;
 };
 
+#if (IS_ENABLED(CONFIG_GD32_EXTI))
```

Comment:
I'm not sure but should it be as below?

```diff
-#if (IS_ENABLED(CONFIG_GD32_EXTI))
+if (IS_ENABLED(CONFIG_GD32_EXTI)) {
...
-#endif
+}
```

Reply:
It's a beautiful solution!

The if-else condition must determine at compile time.
I worry that the intention maybe not have been conveyed to the code reader, adding some comments.